### ISSUE TITLE
Tensorflow build fails due to farmhash build failure on ppc64le

### ISSUE THREAD

**Issue by npanpaliya** [tensorflow/tensorflow#3722]
Tensorflow's external dependency named farmhash fails to build on Ubuntu 14.04 linux ppc64le. If I manually edit the config.guess of farmhash to add a case for ppc64le, then it works. So, I wanted to know how can I contribute this fix back to repo as farmhash is not git cloned by tensorflow, rather a zip archive is downloaded from [farmhash download url](https://github.com/google/farmhash/archive/34c13ddfab0e35422f4c3979f360635a8c050260.zip.)
Even if I generate the a pull request for google/farmhash, how tensorflow will fetch that?

### Environment info
Operating System: Ubuntu 14.04 linux ppc64le

Installed version of CUDA and cuDNN: 7.5
$ ls -l /usr/local/cuda-7.5/lib64/libcud*
-rw-r--r-- 1 root root   326744 Nov  9  2015 /usr/local/cuda-7.5/lib64/libcudadevrt.a
lrwxrwxrwx 1 root root       16 Nov  5  2015 /usr/local/cuda-7.5/lib64/libcudart.so -> libcudart.so.7.5
lrwxrwxrwx 1 root root       19 Nov  5  2015 /usr/local/cuda-7.5/lib64/libcudart.so.7.5 -> libcudart.so.7.5.23
-rwxr-xr-x 1 root root   445192 Nov  5  2015 /usr/local/cuda-7.5/lib64/libcudart.so.7.5.23
-rw-r--r-- 1 root root   902750 Nov  9  2015 /usr/local/cuda-7.5/lib64/libcudart_static.a
lrwxrwxrwx 1 root root       13 Apr 24 20:17 /usr/local/cuda-7.5/lib64/libcudnn.so -> libcudnn.so.5
-rwxr-xr-x 1 root root 60068392 Apr 22 19:18 /usr/local/cuda-7.5/lib64/libcudnn.so.5.0.5
-rw-r--r-- 1 root root 59436124 Apr 22 19:30 /usr/local/cuda-7.5/lib64/libcudnn_static.a

**1. The commit hash (`git rev-parse HEAD`)** - v0.9.0 (problem also seen in the latest master branch)
**2. The output of `bazel version`** 
Build label: head (@f7d9417)
Build target: bazel-out/local_linux-fastbuild/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Tue Aug 9 13:56:57 2016 (1470751017)
Build timestamp: 1470751017
Build timestamp as int: 1470751017

### Steps to reproduce
1. git clone https://github.com/tensorflow/tensorflow
2. cd  tensorflow && git checkout v0.9.0
3. ./configure && bazel build -c opt --config=cuda //tensorflow/tools/pip_package:build_pip_package

### What have you tried?
1. Manually modifying config.guess of farmhash to add a case of ppc64le fixes the problem.

### Logs or other output that would be helpful
ERROR: /home/user/.cache/bazel/_bazel_user/9138b14b59b57ff6760f108ab5f6a562/external/farmhash_archive/BUILD:5:1: Executing genrule @farmhash_archive//:configure failed: bash failed: error executing command /bin/bash -c ... (remaining 1 argument(s) skipped): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
/home/user/.cache/bazel/_bazel_user/9138b14b59b57ff6760f108ab5f6a562/tensorflow_upstream/external/farmhash_archive/farmhash-34c13ddfab0e35422f4c3979f360635a8c050260 /home/user/.cache/bazel/_bazel_user/9138b14b59b57ff6760f108ab5f6a562/tensorflow_upstream
/tmp/tmp.um0vY1ktLs /home/user/.cache/bazel/_bazel_user/9138b14b59b57ff6760f108ab5f6a562/tensorflow_upstream/external/farmhash_archive/farmhash-34c13ddfab0e35422f4c3979f360635a8c050260 /home/user/.cache/bazel/_bazel_user/9138b14b59b57ff6760f108ab5f6a562/tensorflow_upstream
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking build system type... /tmp/tmp.um0vY1ktLs/missing: Unknown `--is-lightweight' option
Try `/tmp/tmp.um0vY1ktLs/missing --help' for more information
configure: WARNING: 'missing' script is too old or missing
./config.guess: unable to guess system type

This script, last modified 2010-08-21, has failed to recognize
the operating system you are using. It is advised that you
download the most up to date version of the config scripts from

  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD
and
  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD

If the version you run (./config.guess) is already up to date, please
send the following data and any information you think might be
pertinent to <config-patches@gnu.org> in order to provide the needed
information to handle your system.

config.guess timestamp = 2010-08-21

uname -m = ppc64le
uname -r = 3.19.0-58-generic
uname -s = Linux
uname -v = #64~14.04.1-Ubuntu SMP Fri Mar 18 19:05:01 UTC 2016

/usr/bin/uname -p =
/bin/uname -X     =

hostinfo               =
/bin/universe          =
/usr/bin/arch -k       =
/bin/arch              =
/usr/bin/oslevel       =
/usr/convex/getsysinfo =

UNAME_MACHINE = ppc64le
UNAME_RELEASE = 3.19.0-58-generic
UNAME_SYSTEM  = Linux
UNAME_VERSION = #64~14.04.1-Ubuntu SMP Fri Mar 18 19:05:01 UTC 2016
configure: error: cannot guess build type; you must specify one

**Comment by vrv** [comment]
We download by git hash: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/workspace.bzl#L35

So if you make a PR to farmhash, and it gets merged, we can update the hash in workspace.bzl and farmhash.BUILD to point to the version that has your commit.

**Comment by npanpaliya** [comment]
Okay. Thanks a lot!

**Comment by cancan101** [comment]
I am seeing this as well on what should be amd64:
```
FarmHash Version 1.1.0

Prefix: '/usr/local'.
Compiler: 'g++ -g -O2'

Now type 'make [<target>]'
  where the optional <target> is:
    all                - build everything
    check              - build and run tests
    install            - install everything

--------------------------------------------------
/root/.cache/bazel/_bazel_root/68a62076e91007a7908bc42a32e4cff9/execroot/tensorflow/external/farmhash_archive/farmhash-34c13ddfab0e35422f4c3979f360635a8c050260 /root/.cache/bazel/_bazel_root/68a62076e91007a7908bc42a32e4cff9/execroot/tensorflow
/root/.cache/bazel/_bazel_root/68a62076e91007a7908bc42a32e4cff9/execroot/tensorflow
/tmp/tmp.0swBa9mWmx/missing: Unknown `--is-lightweight' option
Try `/tmp/tmp.0swBa9mWmx/missing --help' for more information
configure: WARNING: 'missing' script is too old or missing
```

**Comment by npanpaliya** [comment]
Farmhash already has a pending pull request https://github.com/google/farmhash/pull/7 to address this problem. Not sure about the reason for it not yet merged.

**Comment by npanpaliya** [comment]
Farmhash's pull request https://github.com/google/farmhash/pull/8 to detect the ppc64le has been merged to google/farmhash repo.
But looks like the latest tensorflow does not build farmhash as it used to do earlier. So, config.guess and config.sub are not used at all. However, I think it is worth updating farmhash commit hash in the tensorflow/workspace.bzl. 

I have generated a pull request updating the farmhash commit hash in tensorflow https://github.com/tensorflow/tensorflow/pull/4597.
